includes/blocks/course-list/class-sensei-course-list-block.php

```diff
@@ -0,0 +1,56 @@
+<?php
+/**
+ * File containing extra functionalities of extended Sensei_Course_List_Block class.
+ *
+ * @package sensei
+ */
+
+if ( ! defined( 'ABSPATH' ) ) {
+	exit;
+}
+
+/**
+ * Class Sensei_Course_List_Block
+ */
+class Sensei_Course_List_Block {
+
+	/**
+	 * Sensei_Course_List_Block constructor.
+	 */
+	public function __construct() {
+		add_filter( 'render_block', [ $this, 'render_login_form_if_applicable' ], 10, 2 );
+	}
+
+	/**
+	 * Replaces content of Course List block when Show Login Form is set and user is logged out.
+	 *
+	 * @param string $block_content The block content to be rendered.
+	 * @param array  $block          The block to be rendered.
+	 *
+	 * @return string
+	 */
+	public function render_login_form_if_applicable( $block_content, $block ) {
+		if ( 'core/query' !== $block['blockName'] ) {
+			return $block_content;
+		}
+
+		$is_course_list_block = array_key_exists( 'query', $block['attrs'] ) &&
+			'course' === $block['attrs']['query']['postType'] &&
```

Comment:
Should we account for the possibility of `postType` also not existing?

Reply:
Very good observation, added some checks for it [here](https://github.com/Automattic/sensei/pull/6343/commits/590efcbdac8d4f11b5685b5233c251fbc69b2088)